test/Microsoft.AspNet.Mvc.Core.Test/ModelBinding/KeyValuePairModelBinderTest.cs

```diff
@@ -231,7 +232,8 @@ private static ModelBindingContext CreateContext()
                     MetadataProvider = metataProvider,
                     ValidatorProvider = new DataAnnotationsModelValidatorProvider(
                         new TestOptionsManager<MvcDataAnnotationsLocalizationOptions>(),
-                        stringLocalizerFactory: null)
+                        stringLocalizerFactory: null,
+                        validationAttributeAdapterProvider: new ValidationAttributeAdapterProvider())
```

Comment:
validationAttributeAdapterProvider actually IS nullable right now, but I think you're correct that it shouldn't be. Will start checking it and re-arrange.